sonar-core/src/main/java/org/sonar/core/util/FileUtils.java

```diff
@@ -92,12 +102,31 @@ public static boolean deleteQuietly(@Nullable File file) {
     if (file == null) {
       return false;
     }
+    return deleteQuietly(file.toPath());
+  }
+
+  /**
+   * Deletes a file, never throwing an exception. If file is a directory, delete it and all sub-directories.
+   * <p>
+   * The difference between File.delete() and this method are:
+   * <ul>
+   * <li>A directory to be deleted does not have to be empty.</li>
+   * <li>No exceptions are thrown when a file or directory cannot be deleted.</li>
+   * </ul>
+   *
+   * @param file  file or directory to delete, can be {@code null}
+   * @return {@code true} if the file or directory was deleted, otherwise {@code false}
+   */
+  public static boolean deleteQuietly(@Nullable Path path) {
+    if (path == null) {
+      return false;
+    }
 
     try {
-      if (file.isDirectory()) {
-        deleteDirectory(file);
+      if (Files.isDirectory(path)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().isDirectory()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)